File: api/src/main/java/com/beco/api/model/dto/DeliveryOrderNumberDto.java
package com.beco.api.model.dto;

import com.beco.api.model.entity.City;
import com.beco.api.model.entity.Product;
import com.beco.api.model.entity.TransportSupplier;
import jakarta.persistence.*;
import lombok.Data;

import java.util.UUID;

@Data
public class DeliveryOrderNumberDto extends AbstractDto {

    private String uniqueDeliveryOrderNumber;

    private UUID transportSupplierId;
    private String transportSupplierName;

    private UUID customerId;
    private String customerName;

    private UUID cityId;
    private String cityName;

    private UUID productId;
    private String code;
}

File: spa/src/app/core/model/dto/delivery-order-number.dto.ts
import { TransportSupplierDto } from './transport-supplier.dto';
import { CustomerDto } from './customer.dto';
import { CityDto } from './city.dto';
import { ProductDto } from './product.dto';

export interface DeliveryOrderNumberDto {
  deliveryOrderNumberId: string;
  transportSupplier: TransportSupplierDto;
  customer: CustomerDto;
  city: CityDto;
  product: ProductDto;
  uniqueDeliveryOrderNumber: string;
}
